dev/devicelab/lib/tasks/plugin_tests.dart

```diff
@@ -103,4 +105,53 @@ class FlutterProject {
     }
     rmTree(parent);
   }
+
+}
+
+class _FlutterPlugin extends _FlutterProject {
+  _FlutterPlugin(Directory parent, String name) : super(parent, name);
+
+  static Future<_FlutterPlugin> create(
+      Directory directory, List<String> options) async {
+    await inDirectory(directory, () async {
+      await flutter(
+        'create',
+        options: <String>[
+          '--template=plugin',
+          '--org',
+          'io.flutter.devicelab',
+          ...options,
+          'plugintest'
+        ],
+      );
+    });
+    return _FlutterPlugin(directory, 'plugintest');
+  }
+}
+
+class _FlutterApp extends _FlutterProject {
+  _FlutterApp(Directory parent, String name) : super(parent, name);
+
+  static Future<_FlutterApp> create(
+      Directory directory, List<String> options) async {
+    await inDirectory(directory, () async {
+      await flutter(
+        'create',
+        options: <String>[
+          '--template=app',
+          '--org',
+          'io.flutter.devicelab',
+          ...options,
+          'plugintestapp'
+        ],
+      );
+    });
+    return _FlutterApp(directory, 'plugintestapp');
+  }
+
+  Future<void> build(String target) async {
```

Comment:
Curious whether `build` is app specific? Can we build a plugin project too?

Reply:
This only works for apps. For plugins, you have to build the example app.